src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= DEFAULT_LOAD_FACTOR * table.length) {
+            resize();
+        }
+        Node<K, V> node = new Node<>(getHashCode(key), key, value, null);
+        int index = (getHashCode(key) == 0) ? 0 : Math.abs(getHashCode(key)) % table.length;
```

Comment:
```suggestion
        int hash = getHashCode(key);
        Node<K, V> node = new Node<>(hash, key, value, null);
        int index = (hash == 0) ? 0 : Math.abs(hash) % table.length;
```
If there would be complex logic (or that object would be really big, and have a lot of fields) on how to calculate the hashCode of the key, getting this hashCode just once would increase performance

Reply:
changed accordingly